### DISCUSSION TITLE
Production deployment to a VPS

### DISCUSSION THREAD

**Discussion by thansk** [oven-sh/bun#4749] [Q&A]
I am wondering if there is a guide on deploying a bun application without being locked into platforms and their way of doing things. 

My questions are:

1. Can I avoid docker?
2. What happens if I run bun, set up the reverse proxy and after a day, bun crashes for some reason? I know pm2 handles that in the Node ecosystem.
3. Is there a simple way to run bun on all the CPU cores in a VPS?

**Comment by Turbine1991** [comment]
1) The bun website shows how to install bun without docker.

2) The same thing that happens when the bun server is not running.

3) This is more difficult. You already have async programming, currently I'd imagine you'd have to use a threading library and code your app to do multi threading. There'll be no magic flag.

**Reply by thansk** [reply]
> There'll be no magic flag.

Are you part of Oven.sh?

**Reply by mbyrne00** [reply]
@Turbine1991 - can you please point to the part on the website where (1) is explained, please? 

In my case I'd like to be able to run in an environment where we can't do an `npm i -g` first. Environments like Google App Engine, Google Cloud Functions, and other Serverless/PaaS environments that run node. Just want to see what the limitations are and if this is viable.

**Reply by hepidad** [reply]
@mbyrne00 For (1) maybe [this](https://bun.com/docs/installation#macos-%26-linux) ?
